android/app/src/main/java/com/zulipmobile/MainActivity.kt

```diff
@@ -11,6 +11,25 @@ import com.swmansion.gesturehandler.react.RNGestureHandlerEnabledRootView;
 import com.zulipmobile.notifications.*
 import com.zulipmobile.sharing.maybeHandleShareIntent
 
+/* Returns true just if we did handle the intent. */
+fun maybeHandleOpenNotificationIntent(intent: Intent): Boolean {
+    // We handle intents to launch MainActivity on tapping notifications
+    if (
+        intent.action == Intent.ACTION_VIEW
+        // The web-auth intent's action is also VIEW; leave that for RN
+        //   to handle.
+        && intent.hasExtra(EXTRA_NOTIFICATION_DATA)
+    ) {
+        val data = intent.getBundleExtra(EXTRA_NOTIFICATION_DATA)
+        logNotificationData("notif opened", data!!)
```

Comment:
android notif: Move a log line into MainActivity's lifecycle methods

This commit is kind of confusing because it's rather a lot more code than just moving a log line sounds like it should be:
 2 files changed, 26 insertions(+), 4 deletions(-)

The added plumbing makes sense because of the real work we have it do in the next commit. I think the clearest thing is probably to squash that one into this. The result is only slightly bigger than this already is:
 2 files changed, 26 insertions(+), 6 deletions(-)

and it's easier to understand as changing where we do the real work, with the log line coming along for the ride.